### PR TITLE
fix: react-combobox should close on blur after clicking expand icon

### PR DESCRIPTION
**Pull request by smhigley**
Addresses an item in #25724

There was a bug in how the expand icon click interacted with close-on-blur -- it needs to not close on blur when clicking to close (which would cause it to close then open again), but should close on blur after clicking to open.

PR adds fix + test

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/Avatar-mount_base.html">1286</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/Avatar-mount.html">1268</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/Button-mount_base.html">931</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/Button-mount.html">920</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/FluentProvider-mount_base.html">1484</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/FluentProvider-mount.html">1478</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/FluentProviderWithTheme-mount_base.html">568</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/FluentProviderWithTheme-mount.html">589</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">537</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">545</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">582</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">572</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/MakeStyles-mount_base.html">1972</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/MakeStyles-mount.html">1953</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/SpinButton-mount_base.html">2319</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25771/perf-test-react-components/SpinButton-mount.html">2322</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                             |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                               Change |
| :---------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `78.244 kB`<br />`25.015 kB` | `78.249 kB`<br />`25.018 kB` | `5 B` <br />`3 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `77.531 kB`<br />`24.927 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.975 kB`<br />`17.687 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `190.427 kB`<br />`53.087 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |   `33.48 kB`<br />`11.037 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/cd3cdee917b09bc31005de0e789609e9b55968bd'>cd3cdee917b09bc31005de0e789609e9b55968bd</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 2 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-tool-fluent/index.html?organization=uifabric&project=fabricpublic&prId=25771&commitId=2f3361c83accc9b207e121b2f36003084010ac92&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1749 screenshots unchanged, 0 screenshots with different dimensions and 2 screenshots with visible difference.


  <details><summary><b>unknown</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.size+active+badge.normal.chromium.png|11| Changed|
|Avatar Converged.size+active+ring-shadow.normal.chromium.png|5| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

**Review by micahgodbolt — APPROVED** [review]

File: packages/react-components/react-combobox/src/components/Combobox/Combobox.test.tsx [hunk 1]
```diff
@@ -133,6 +133,25 @@ describe('Combobox', () => {
     expect(queryByRole('listbox')).toBeNull();
   });
 
+  it('closes the popup on blur/outside click', () => {
```
Comment (micahgodbolt) [inline packages/react-components/react-combobox/src/components/Combobox/Combobox.test.tsx:136]:
```suggestion
  it('closes the popup on blur/outside click after clicking on the expand icon', () => {
```

is this the problem you found?

Comment (smhigley) [reply packages/react-components/react-combobox/src/components/Combobox/Combobox.test.tsx:136]: yup!

**Pull request merged**